### ISSUE TITLE
Using old version to satisfy dependency

### ISSUE THREAD

**Issue by kontura** [openSUSE/libsolv#514]
The following test case passes with current upstream:
```t
repo @System 0 testtags <inline>
#>=Pkg: crypto-policies 3 1 noarch
#>=Obs: crypto-policies < 2-1
#>=Rec: crypto-policies-scripts

repo available 0 testtags <inline>
#>=Pkg: java-11-openjdk-headless 1 1 x86_64
#>=Req: nss
#>
#>=Pkg: nss 1 1 x86_64
#>=Prv: nss = 1-1
#>
#>=Pkg: nss 2 1 x86_64
#>=Req: /usr/bin/update-crypto-policies
#>=Prv: nss = 2-1
#>
#>=Pkg: crypto-policies 1 2 noarch
#>=Prv: /usr/bin/update-crypto-policies
#>
#>=Pkg: crypto-policies-scripts 3 1 noarch
#>=Prv: /usr/bin/update-crypto-policies
#>=Req: crypto-policies = 3-1

# crypto-policies-scripts was split from crypto-policies

system x86_64 rpm @System
poolflags implicitobsoleteusescolors whatprovideswithdisabled
solverflags allowdowngrade allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes

job install oneof java-11-openjdk-headless-1-1.x86_64@available [forcebest,setevr,setarch]

result transaction,problems <inline>
#>install java-11-openjdk-headless-1-1.x86_64@available
#>install nss-1-1.x86_64@available
```

While this solution is valid it is unexpected. Doing an additional upgrade after this uses the new `nss-2-1`.

This behavior was changed by patch https://github.com/openSUSE/libsolv/commit/1da9bef88dd269055cbd7eda2f3572963d6d9b64 and before it it worked as expected:
```t
result transaction,problems <inline>
#>install java-11-openjdk-headless-1-1.x86_64@available
#>install nss-2-1.x86_64@available
#>install crypto-policies-scripts-3-1.noarch@available
```

In case it would be useful I have also noticed that when the `allowdowngrade` is removed or ` strongrecommends` is added it works correctly.

Thank you for any help.

**Comment by mlschroe** [comment]
Yes, it's weird. It happens because the solver does not like to pull in the new "crypto-policies-scripts" package. I'll have a look.

**Comment by mlschroe** [comment]
It's caused by the "allowdowngrade" flag...

**Comment by mlschroe** [comment]
Fixed with commit 2c85ed581422e072ad95119f3d7dc19eb45f29ac

I hope that this doesn't introduce new regressions. We've got a lot of testcases for choice rules, so it isn't that likely.
